.jenkins/pytorch/test.sh

```diff
@@ -55,7 +55,8 @@ else
   export PYTORCH_TEST_SKIP_NOARCH=1
 fi
 
-if [[ -n "$IN_PULL_REQUEST" ]]; then
+if [[ -n "$IN_PULL_REQUEST" ]] && [[ -z "$CI_MASTER" || "$CI_MASTER" == "false" ]]; then
+  # skip expensive checks when on PR and CI_MASTER flag is not set
   export PYTORCH_TEST_SKIP_CUDA_MEM_LEAK_CHECK=1
```

Comment:
looks fine but just curious, what's the situation where `IN_PULL_REQUEST` isn't always the opposite of `CI_MASTER`?

Reply:
it's complicated: https://github.com/pytorch/pytorch/issues/60746